roles/openshift_gcp/templates/provision_ssh.j2.sh

```diff
@@ -23,13 +23,34 @@ if [[ -n "{{ openshift_gcp_ssh_private_key }}" ]]; then
         pub_key=$(cut -d ' ' -f 2 <  "${pub_file}")
     fi
     key_tmp_file='/tmp/ocp-gce-keys'
-    if ! gcloud --project "{{ openshift_gcp_project }}" compute project-info describe | grep -q "$pub_key"; then
-        if gcloud --project "{{ openshift_gcp_project }}" compute project-info describe | grep -q ssh-rsa; then
-            gcloud --project "{{ openshift_gcp_project }}" compute project-info describe | grep ssh-rsa | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//' -e 's/value: //' > "$key_tmp_file"
+    echo -n 'cloud-user:' >> "$key_tmp_file"
+    cat "${pub_file}" >> "$key_tmp_file"
+
+    # Set up ssh keys for the builder instance.
+    (
+        build_instance="{{ openshift_gcp_prefix }}build-image-instance"
+        if ! metadata=$(gcloud --project "{{ openshift_gcp_project }}" compute instances describe "${build_instance}" --zone "{{ openshift_gcp_zone }}" --format='value[](metadata.items.ssh-keys)'); then
+            exit 0
         fi
-        echo -n 'cloud-user:' >> "$key_tmp_file"
-        cat "${pub_file}" >> "$key_tmp_file"
-        gcloud --project "{{ openshift_gcp_project }}" compute project-info add-metadata --metadata-from-file "sshKeys=${key_tmp_file}"
-        rm -f "$key_tmp_file"
-    fi
+        if ! echo "${metadata}" | grep -q "${pub_key}"; then
+            gcloud --project "{{ openshift_gcp_project }}" compute instances add-metadata "${build_instance}" --zone "{{ openshift_gcp_zone }}" --metadata-from-file ssh-keys="${key_tmp_file}"
+        fi
+    ) &
+
+    # Set up ssh keys for all instances in all groups.
+    {% for node_group in openshift_gcp_node_group_config %}
```

Comment:
This won't work with autoscaling?

Reply:
Well, I definitely didn't consider that... are there any tests which are impacted? The current implementation seems almost constantly broken for me (perhaps depending on the amount of cluster churn at a given time triggering the global variable length limit issue).

Will have to think more about this in terms of autoscaling...